packages/bee-q/src/components/textarea/bq-textarea.tsx

```diff
@@ -0,0 +1,298 @@
+import { Component, Element, Event, EventEmitter, h, Prop, State } from '@stencil/core';
+
+import { TTextareaAutoCapitalize, TTextareaWrap } from './bq-textarea.types';
+import { debounce, hasSlotContent, isHTMLElement, TDebounce } from '../../shared/utils';
+import { TInputValidation } from '../input/bq-input.types';
+
+@Component({
+  tag: 'bq-textarea',
+  styleUrl: './scss/bq-textarea.scss',
+  shadow: {
+    delegatesFocus: true,
+  },
+})
+export class BqTextarea {
+  // Own Properties
+  // ====================
+
+  private debounceBqInput: TDebounce<void>;
+  private fallbackId = 'textarea';
+
+  private helperTextElem?: HTMLElement;
+  private labelElem?: HTMLLabelElement;
+  private textarea: HTMLTextAreaElement;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqTextareaElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  @State() private hasHelperText = false;
+  @State() private hasLabel = false;
+  @State() private numberOfCharacters: number = 0;
```

Comment:
Would be better to create a getter to retrieve `numberOfCharacters`. Or we add a watch for `value` property change. Otherwise state won't be in sync when changed externally 
```typescript
private get numberOfCharacters() {
    if (!this.maxlength || !this.textarea) return 0;

    return this.value.length;
}
```

Reply:
Solved! Thx 🙏🏼